### ISSUE TITLE
OpenBoard does not work with drawing tablets

### ISSUE THREAD

**Issue by lrdegeest** [OpenBoard-org/OpenBoard#139]
Unfortunately, OpenBoard is not stable with drawing tablets that require digital pens. Sometimes when I use the pencil tool, in the middle of drawing something, or annotating an image or text, OpenBoard will randomly switch to the eraser, and will not allow me to switch back to the pencil.

**Comment by obook** [comment]
What kind of drawing tablet is ? I Use OpenBoard with Wacom Bamboo and Wacom Intuos whiyhout any problem.

**Comment by fsiegert** [comment]
I am also having problems with the drawing tablet built into the Lenovo Thinkpad X1 Yogo (which I think is a Wacom): OpenBoard is not able to disable drawing by touch (e.g. from fingers/palm), but instead will draw lines from the beginning of a touch/tap to the end. Maybe it would be possible to take some of the code for this from xournal (http://xournal.sourceforge.net/) where this works perfectly well?

**Comment by obook** [comment]
You can remove palm touch by disabling "finger-touch" driver(s) in Devices Manager under Windows.

**Comment by fsiegert** [comment]
Hi @obook ,

thanks for the advise! This might be a temporary workaround, but I would not want to generally disable touch (e.g. it is still useful to use it for the rest of the OpenBoard interface, or any other programs), but only disable the drawing. That's why I think a system-wide setting won't be helpful and it has to be an intelligent option in the software.

Maybe I should file a separate wishlist ticket for this. I just wanted to hear first here whether others with drawing tablets have the same problem or whether I'm missing something, but this is probably leading the original ticket a bit far off-topic now.

Best,  
Frank

**Comment by poinck** [comment]
@fsiegert I have the same problem. Bamboo Wacom is not recognized when it is in other wacom-capable programs. I am using it under Gnome with wayland and have it installed via flatpak from flathub.

**Comment by PeterPorker3** [comment]
Hello, I see this is an old issue and there hasn't been a lot of activity on it, but I figured I would give it a shot. I have a Dell Latitude 7275 tablet (which has a Wacom pen digitizer) running the latest stable version of Ubuntu 20.04 LTS. Its basically a fresh install, the only thing I have installed is OpenBoard 1.6.1,  OnBoard, and (later on) the latest open source Wacom software I compiled from here: https://github.com/linuxwacom/input-wacom I believe. 

At first I was using the default X server mode and the included Wacom drivers and I immediately ran into multiple issues. The most noticeable was that while in use, it would sometimes switch to the eraser (like @lrdegeest mentioned) and wouldn't let me click any buttons at all for 15-20 seconds, and only after that would it allow me to use the OpenBoard UI to switch back to the pen or show the desktop. It would also make the device unresponsive at times only when OpenBoard is running, which is strange for a machine with an i5 and an SSD. Sometimes it would even lock the whole system up, with the mouse cursor able to move but no other inputs are able to do anything. Also, sometimes for a few soconds the pen would behave sporadically, but this was pretty rare. In general it wasn' t a great experience. 

Then I tried running Ubuntu in Wayland mode and things got a bit better. The device was in general more responsive, even when Openboard was running. The freezing issue didn't occur as often, but still happens from time to time, but interestingly only when OpenBoard is autosaving or performing file operations. The eraser issue seems to have gone away too. The pen would still sometimes behave sporadically, but installing the latest input-wacom kernel drives seems to have fixed this issue on Wayland at least (didn't try it in x). However there is a new issue: Sometimes (the only way i found I can consistently replicate it is when I try to snip part of the screen while the desktop is shown) it would fill the screen with black (while still showing the floating tools box), and any screenshot I try to make would just be a blank black image.

Palm rejection has never worked reliably in any configuration for me. I think this is down to Ubuntu's limited settings and options for wacom tablets, as this should be handled by the wacom driver and not by OpenBoard (I assume). If anyone knows how to improve this without disabling the touch functionality outright, please let me know!

So from what I can tell, building the latest input-wacom kernel drivers and using Ubuntu Wayland seems to produce more stable behavior, despite the black screen issue. However I'm curious if someone had already found a solution to any of these issues. It would be nice to have a more reliable way to take digital notes. Even if not, I hope this description of my experience helps in troubleshooting these issues. 

However there are some caveats to what I have talked about that I would like to mention here:

-Some of these issues could be a problem with my OS. Like I said, this is pretty much a fresh install, but there could still be issues with Ubuntu itself. I don't have the time to test other distros, so I can't validate how much of this is Ubuntu wierdness and how much is OpenBoard wierdness. However before using Ubuntu, I did briefly try OpenBoard on Kubuntu 20.04 on the same device and I don't remember having any of the major issues I listed above. However I never use it extensively, I switched away because KDE just wasn't very tablet friendly and wasn't super responsive on the machine. 

-There could be compatibility issues with some piece of software in the chain and my specific hardware. This model of tablet seems to be pretty uncommon, so an incompatibility with something is not out of the question as official software support for this specific model of laptop in ubuntu is pretty much nonexistent. However those Lenovo X1 Tablets seem to be more common and more linux friendly, yet there have been reports of some similar issues (mentioned above)

-There could be something physically wrong with my hardware. The freezes are rather strange and to be honest I have no idea if they are related to OpenBoard and Ubuntu. Because they seem to happen most often during file operations, I suppose its possible something is wrong with my SSD, but SMART says its healthy and I haven't really had any issues using the device with other programs. The tablet in general is in pretty rough shape thought so I wouldn't be surprised if it wasn't 100% stable.
